### DISCUSSION TITLE
Nested SSH

### DISCUSSION THREAD

**Discussion by danchitnis** [Xpra-org/xpra#3795] [Q&A] [answered]
Hello, I am trying to connect to an xpra server behind a firewall, and access is only available via an organization level SSH server. So I have first to do `ssh gateway`, then in that gateway, I have to do `ssh my-server`. Both these ssh connections require manual typing of the password. How can I get xpra working with this nested connection?

**Comment by danchitnis** [comment, marked as answer]
For anyone trying to do this first ssh as usual via the gateway to your `my_server` and start the xpra server:

```bash
xpra start :100 --start=xclock
```

then on the client outside the firewall run:

```bash
xpra attach --ssh="ssh -J user@gateway.com" ssh://user@my-server.com/100
```

You have to type in your password for each server, and then should attach to the xpra session on `my_server`

**Comment by totaam** [comment]
There are many ways of doing this.
Including:
* https://github.com/Xpra-org/xpra/issues/2041
* ssh proxy command
* https://github.com/Xpra-org/xpra/issues/3745 (not implemented yet - though looks simple enough)
etc..
